src/horus_api.c

```diff
@@ -368,24 +399,12 @@ int horus_rx(struct horus *hstates, char ascii_out[], short demod_in[]) {
     for(i=0,j=Nbits; j<rx_bits_len; i++,j++) {
         hstates->rx_bits[i] = hstates->rx_bits[j];
     }
-                   
+
     /* demodulate latest bits */
 
-    /* Note: allocating this array as an automatic variable caused OSX to
-       "Bus Error 10" (segfault), so lets malloc() it.  TODO: A real
-       short sample option for fsk_demod() would be useful */
-    
-    COMP *demod_in_comp = (COMP*)malloc(sizeof(COMP)*hstates->fsk->nin);
-    
-    for (i=0; i<hstates->fsk->nin; i++) {
-        demod_in_comp[i].real = demod_in[i];
-        demod_in_comp[i].imag = 0;
-    }
     fsk_demod(hstates->fsk, &hstates->rx_bits[rx_bits_len-Nbits], demod_in_comp);
```

Comment:
This code was reqd to prevent problems with the OSX build, as per the comments.  It would be wise to keep it in horus_rx() above.

@darksidelemm - you debugged this problem originally, what do you think?

Reply:
Yes, that should probably be kept. Caused a lot of grief.